src/SFA.DAS.EmployerFinance.Jobs/Program.cs

```diff
@@ -7,64 +7,72 @@
 using Microsoft.Extensions.Logging;
 using NLog.Extensions.Logging;
 using SFA.DAS.AutoConfiguration;
+using SFA.DAS.EmployerFinance.Configuration;
 using SFA.DAS.EmployerFinance.Jobs.DependencyResolution;
 using SFA.DAS.EmployerFinance.Startup;
+using StructureMap;
 
 namespace SFA.DAS.EmployerFinance.Jobs
 {
     public static class Program
     {
-        private const string AppSettingFilePath = "appsettings.json";
-
         public static async Task Main()
         {
             ServicePointManager.DefaultConnectionLimit = 50;
 
             using (var container = IoC.Initialize())
             {
-                var startup = container.GetInstance<IStartup>();
+                var environmentService = container.GetInstance<IEnvironmentService>();
 
-                await startup.StartAsync();
+                var environmentSettings = HostEnvironmentSettingsFactory.Create(environmentService);
 
-                var jobActivator = new StructureMapJobActivator(container);
+                var host = CreateHost(container, environmentSettings);
 
-                var environmentService = container.GetInstance<IEnvironmentService>();
+                await RunHost(container, host);
+            }
+        }
+        
+          private static async Task RunHost(IContainer container, IHost host)
+        {
+            var startup = container.GetInstance<IStartup>();
 
-                var environmentName = environmentService.IsCurrent(DasEnv.LOCAL)
-                    ? EnvironmentName.Development
-                    : EnvironmentName.Production;
+            await startup.StartAsync();
 
-                var hostBuilder = new HostBuilder()
-                    .UseEnvironment(environmentName)
-                    .ConfigureWebJobs(builder =>
-                    {
-                        builder.AddAzureStorageCoreServices()
-                               .AddTimers();
-                    })
-                    .ConfigureAppConfiguration(builder =>
-                    {
-                        builder.AddJsonFile(AppSettingFilePath);
-                    })
-                    .ConfigureLogging(builder => 
-                    { 
-                        builder.SetMinimumLevel(LogLevel.Debug)
-                               .AddNLog();
-                    })
-                    .ConfigureServices(collection =>
-                    {
-                        collection.AddSingleton<IJobActivator>(jobActivator);
-                    })
-                    .UseConsoleLifetime();
+            using (host)
+            {
+                await host.RunAsync();
+            }
 
-                var host = hostBuilder.Build();
+            await startup.StopAsync();
+        }
 
-                using (host)
+        private static IHost CreateHost(IContainer container, HostEnvironmentSettings environmentSettings)
```

Comment:
In both the `Jobs` and `MessageHandler` webjobs the startup code looks like its exploded over 100 lines across multiple methods and files. Can it be condensed? It should fit into ~20 lines e.g.

```c#
ServicePointManager.DefaultConnectionLimit = 50;

using (var container = IoC.Initialize())
{
    var environmentService = container.GetInstance<IEnvironmentService>();
    var environmentName = environmentService.IsCurrent(DasEnv.LOCAL) ? EnvironmentName.Development : EnvironmentName.Production;
    var jobActivator = new StructureMapJobActivator(container);
    var startup = container.GetInstance<IStartup>();
    
    var host = new HostBuilder()
        .UseEnvironment(environmentName)
        .ConfigureWebJobs(b => b.AddAzureStorageCoreServices().AddTimers())
        .ConfigureAppConfiguration(b => b.AddJsonFile("appsettings.json").AddJsonFile($"appsettings.{environmentName}.json", true))
        .ConfigureLogging(b => b.SetMinimumLevel(LogLevel.Debug).AddNLog())
        .ConfigureServices(c => c.AddSingleton<IJobActivator>(jobActivator))
        .UseConsoleLifetime()
        .Build();

    await startup.StartAsync();
    
    using (host)
    {
        await host.RunAsync();
    }

    await startup.StopAsync();
```

Reply:
I spoke with Michael Young about this and also mentioned it to Phil. The issue is that a lot is going on in those 20 lines above and we wanted to split the logic up across multiple methods to break it down. Having multiple builder calls on a single line does reduce the line count but also makes it harder to read especially if we end up adding more builder calls. I'm happy to change the code to reflect the above if you think breaking it down isn't useful?

Reply:
spoken with Elliot and will use the above approach